htdocs/core/lib/ajax.lib.php

```diff
@@ -517,7 +517,7 @@ function ajax_combobox($htmlname, $events = array(), $minLengthToAutocomplete =
 	 					/* Code to add class of origin OPTION propagated to the new select2 <li> tag */
 						if (data.element) { $(container).addClass($(data.element).attr("class")); }
 						//console.log("data html is "+$(data.element).attr("data-html"));
-						if (data.id == '.((int) $idforemptyvalue).' && $(data.element).attr("data-html") == undefined) {
+						if (data.id == "'.dol_escape_js($idforemptyvalue, 2).'" && $(data.element).attr("data-html") == undefined) {
```

Comment:
The char to use for js string must be ' (this is the char escaped by dol_escape_js).
To should be
`if (data.id == \''.dol_escape_js($idforemptyvalue, 2).'\' && $(data.element).attr("data-html") == undefined) {`
Can you test this ?